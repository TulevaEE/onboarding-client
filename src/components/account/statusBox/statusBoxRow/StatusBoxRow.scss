@import '../../../variables';

.status-box-row li {
  list-style: none;
}

.status-box-row ul {
  padding: 0;
  margin-bottom: 0;
}

.status-box-row .btn {
  min-width: 161px;
}

.fa.fa-check {
  color: $green;
}

.fa.fa-times {
  color: $red;
}

.fa.fa-exclamation-triangle {
  color: $yellow-dark;
}
