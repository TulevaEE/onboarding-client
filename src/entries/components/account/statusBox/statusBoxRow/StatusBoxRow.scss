@import "../../../variables";

li.status-box-row {
  list-style: none;
}

ul.status-box-row {
  padding: 0;
  margin-bottom: 0;
}

.fa.fa-check {
  color: $green;
}

.fa.fa-times {
  color: $red;
}
