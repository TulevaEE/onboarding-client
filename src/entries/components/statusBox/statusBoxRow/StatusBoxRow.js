import React from 'react';
import { PropTypes as Types } from 'prop-types';
import { Message } from 'retranslate';

const StatusBoxRow = ({ name, showAction, ok }) => {
  const displayName = <Message>{name}</Message>;
  const displayAction = 'XXXXXXX tbd siia nupp';
  return (
    <div className="row tv-table__row py-2">
      <div className="col-12 col-sm">
        {ok ? 'âœ”' : 'ðŸ—™'} <b>{displayName}</b>
      </div>
      <div className="col-12 col-sm text-sm-right">{showAction ? <b>{displayAction}</b> : ''}</div>
    </div>
  );
};

StatusBoxRow.defaultProps = {
  name: '',
  showAction: false,
  ok: false,
};

StatusBoxRow.propTypes = {
  name: Types.oneOfType([Types.node, Types.string]),
  showAction: Types.bool,
  ok: Types.bool,
};

export default StatusBoxRow;
