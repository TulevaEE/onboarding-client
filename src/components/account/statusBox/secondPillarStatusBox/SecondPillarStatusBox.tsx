import React from 'react';
import { FormattedMessage } from 'react-intl';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import moment from 'moment';
import StatusBoxRow from '../statusBoxRow';
import { useMandateDeadlines, usePendingApplications } from '../../../common/apiHooks';
import {
  Application,
  ApplicationType,
  Conversion,
  Fund,
  MandateDeadlines,
  SourceFund,
} from '../../../common/apiModels';
import { State } from '../../../../types';
import InfoTooltip from '../../../common/infoTooltip';
import { isTuleva } from '../../../common/utils';
import { getValueSum } from '../../AccountStatement/fundSelector';
import Euro from '../../../common/Euro';
import { formatDateTime, formatDateYear } from '../../../common/dateFormatter';
import deadline from './deadline.svg';
import euro from './euro.svg';
import basket from './basket.svg';

export interface Props {
  loading: boolean;
  conversion: Conversion;
  sourceFunds: SourceFund[];
  targetFunds: Fund[];
  secondPillarPikNumber: string | null;
  secondPillarActive: boolean;
  secondPillarPaymentRate: number;
}

export const SecondPillarStatusBox: React.FC<Props> = ({
  loading,
  conversion,
  sourceFunds,
  targetFunds,
  secondPillarPikNumber,
  secondPillarActive,
  secondPillarPaymentRate,
}: Props) => {
  const activeFund = sourceFunds.find((fund) => fund.activeFund);
  const { data: mandateDeadlines } = useMandateDeadlines();

  if (!secondPillarActive || !activeFund) {
    return (
      <StatusBoxRow
        error
        showAction={!loading}
        name={<FormattedMessage id="account.status.choice.pillar.second" />}
        lines={[<FormattedMessage id="account.status.choice.pillar.second.missing.label" />]}
      />
    );
  }

  const pendingWithdrawal = usePendingWithdrawalApplication();
  if (conversion.pendingWithdrawal && pendingWithdrawal) {
    return (
      <StatusBoxRow
        error
        showAction={!loading}
        name={<FormattedMessage id="account.status.choice.pillar.second" />}
        lines={[<FormattedMessage id="account.status.choice.pillar.second.withdraw" />]}
      >
        <Link to={`/applications/${pendingWithdrawal.id}/cancellation`} className="btn btn-primary">
          <FormattedMessage id="account.status.choice.pillar.second.withdraw.cancel" />
        </Link>
      </StatusBoxRow>
    );
  }

  if (secondPillarPikNumber) {
    return (
      <StatusBoxRow
        showAction={!loading}
        name={<FormattedMessage id="account.status.choice.pillar.second" />}
        lines={[
          <FormattedMessage
            id="account.status.choice.pillar.second.pik"
            values={{ secondPillarPikNumber }}
          />,
        ]}
      />
    );
  }

  const isPartiallyConverted = conversion.selectionPartial || conversion.transfersPartial;
  if (!isPartiallyConverted) {
    if (conversion.weightedAverageFee > 0.005) {
      return highFee(loading, conversion, sourceFunds, targetFunds, mandateDeadlines);
    }
    return (
      <StatusBoxRow
        ok
        showAction={!loading}
        name={<FormattedMessage id="account.status.choice.pillar.second" />}
        lines={[
          <>
            <FormattedMessage id="account.status.choice.lowFee.label" />
            <InfoTooltip name="second-pillar-tooltip">
              <FormattedMessage id="account.status.choice.lowFee.description" />
            </InfoTooltip>
          </>,
        ]}
      >
        <Link to="/2nd-pillar-flow" className="btn btn-light">
          <FormattedMessage id="account.status.choice.join.tuleva.2" />
        </Link>
      </StatusBoxRow>
    );
  }

  const isFullyConverted = conversion.selectionComplete && conversion.transfersComplete;
  if (!isFullyConverted) {
    if (conversion.weightedAverageFee > 0.005) {
      return highFee(loading, conversion, sourceFunds, targetFunds, mandateDeadlines);
    }
    return (
      <StatusBoxRow
        error
        showAction={!loading}
        name={<FormattedMessage id="account.status.choice.pillar.second" />}
        lines={[
          <FormattedMessage id="account.status.choice.pillar.second.transferIncomplete" />,
          ...(mandateDeadlines && isPeriodEnding(mandateDeadlines)
            ? [periodEndingMessage(mandateDeadlines)]
            : []),
        ]}
      >
        <Link to="/2nd-pillar-flow" className="btn btn-primary">
          <FormattedMessage id="account.status.choice.transfer.tuleva.2" />
        </Link>
      </StatusBoxRow>
    );
  }

  if (secondPillarPaymentRate < 4) {
    return (
      <StatusBoxRow
        ok
        showAction={!loading}
        name={<FormattedMessage id="account.status.choice.pillar.second" />}
        lines={[
          <FormattedMessage
            id="account.status.choice.lowFee.index.2.label"
            values={{
              paymentRate: secondPillarPaymentRate,
            }}
          />,
          <small className="text-muted">
            <FormattedMessage
              id="account.status.choice.pillar.second.paymentRate.comment"
              values={{
                b: (chunks: string) => <b>{chunks}</b>,
              }}
            />
          </small>,
        ]}
      >
        <Link to="/2nd-pillar-payment-rate" className="btn btn-primary">
          <FormattedMessage id="account.status.choice.paymentRate.increase" />
        </Link>
      </StatusBoxRow>
    );
  }

  return (
    <StatusBoxRow
      ok
      name={<FormattedMessage id="account.status.choice.pillar.second" />}
      showAction={!loading}
      lines={[
        <FormattedMessage
          id="account.status.choice.lowFee.index.2.label"
          values={{
            paymentRate: 2,
          }}
        />,
        <small className="text-muted">
          <FormattedMessage
            id="account.status.choice.lowFee.index.2.description"
            values={{
              paymentRate: secondPillarPaymentRate,
              paymentRateFulfillmentDate: formatDateYear(
                mandateDeadlines?.paymentRateFulfillmentDate,
              ),
            }}
          />
        </small>,
      ]}
    >
      {secondPillarPaymentRate < 6 && (
        <Link to="/2nd-pillar-payment-rate" className="btn btn-primary">
          <FormattedMessage id="account.status.choice.paymentRate.increase" />
        </Link>
      )}
    </StatusBoxRow>
  );
};

function feeComparison(currentFeesEuro: number, tulevaFeesEuro: number) {
  return (
    <small className="text-muted">
      <FormattedMessage
        id="account.status.choice.highFee.comment"
        values={{
          currentFeesEuro: (
            <Euro amount={currentFeesEuro} className="text-danger" fractionDigits={0} />
          ),
          tulevaFeesEuro: (
            <Euro amount={tulevaFeesEuro} className="text-success" fractionDigits={0} />
          ),
        }}
      />
    </small>
  );
}

const TinyCard = ({ title, text, img }: { title: JSX.Element; text: JSX.Element; img: string }) => (
  <div className="col-md-4 my-3 my-md-0">
    <div className="d-flex align-items-center">
      <div className="mr-3">
        <img width={36} src={img} alt="" />
      </div>
      <div>
        <h5 className="card-title mb-1">{title}</h5>
        <small className="card-text">{text}</small>
      </div>
    </div>
  </div>
);

function highFee(
  loading: boolean,
  conversion: Conversion,
  sourceFunds: SourceFund[],
  targetFunds: Fund[],
  mandateDeadlines: MandateDeadlines | undefined,
) {
  const value = getValueSum(sourceFunds);
  const currentFeesEuro = conversion.weightedAverageFee * value;
  const tulevaTargetFunds =
    targetFunds && targetFunds.length && targetFunds.filter((fund) => isTuleva(fund));
  const defaultTargetFund =
    tulevaTargetFunds && tulevaTargetFunds.length ? tulevaTargetFunds[0] : null;
  const tulevaFeesPercent = defaultTargetFund?.ongoingChargesFigure || 0;
  const tulevaFeesEuro = tulevaFeesPercent * value;

  return (
    <StatusBoxRow
      error
      showAction={!loading}
      name={<FormattedMessage id="account.status.choice.pillar.second" />}
      lines={[
        <>
          <FormattedMessage id="account.status.choice.highFee.label" />
          <InfoTooltip name="second-pillar-tooltip">
            <FormattedMessage id="account.status.choice.highFee.description" />
          </InfoTooltip>
        </>,
        ...(currentFeesEuro >= 0.5 && tulevaFeesEuro >= 0.5 && currentFeesEuro - tulevaFeesEuro >= 1
          ? [feeComparison(currentFeesEuro, tulevaFeesEuro)]
          : []),
      ]}
      extraBottom={
        <div className="card card-primary border-0 mt-3 ml-3 ml-md-5 mr-3 my-2 px-4 py-3">
          <div className="row">
            <TinyCard
              img={euro}
              title={<FormattedMessage id="account.status.choice.pillar.second.feecard.title" />}
              text={
                <FormattedMessage
                  id="account.status.choice.pillar.second.feecard.text"
                  values={{
                    b: (chunks: string) => <b>{chunks}</b>,
                  }}
                />
              }
            />
            <TinyCard
              img={basket}
              title={<FormattedMessage id="account.status.choice.pillar.second.basketcard.title" />}
              text={
                <>
                  <FormattedMessage
                    id="account.status.choice.pillar.second.basketcard.text"
                    values={{
                      b: (chunks: string) => <b>{chunks}</b>,
                    }}
                  />
                  <InfoTooltip name="diversification-tooltip">
                    <FormattedMessage id="account.status.choice.pillar.second.basketcard.tooltip" />
                  </InfoTooltip>
                </>
              }
            />
            <TinyCard
              img={deadline}
              title={
                <FormattedMessage id="account.status.choice.pillar.second.deadlinecard.title" />
              }
              text={
                <FormattedMessage
                  id="account.status.choice.pillar.second.deadlinecard.text"
                  values={{
                    periodEnding: formatDateTime(mandateDeadlines?.periodEnding),
                    b: (chunks: string) => <b className="text-nowrap">{chunks}</b>,
                  }}
                />
              }
            />
          </div>
        </div>
      }
    >
      <Link to="/2nd-pillar-flow" className="btn btn-primary">
        <FormattedMessage id="account.status.choice.bring.to.tuleva" />
      </Link>
    </StatusBoxRow>
  );
}

function isPeriodEnding(mandateDeadlines: MandateDeadlines | undefined) {
  return mandateDeadlines && moment(mandateDeadlines.periodEnding).diff(moment(), 'days') <= 30;
}

function periodEndingMessage(mandateDeadlines: MandateDeadlines | undefined) {
  return (
    mandateDeadlines && (
      <small className="text-muted">
        <FormattedMessage
          id="select.sources.select.all.deadline"
          values={{
            periodEnding: formatDateTime(mandateDeadlines?.periodEnding),
            b: (chunks: string) => <b className="text-nowrap">{chunks}</b>,
          }}
        />
      </small>
    )
  );
}

const usePendingWithdrawalApplication = (): Application | undefined =>
  usePendingApplications().data?.find(
    (application) =>
      application.type === ApplicationType.WITHDRAWAL ||
      application.type === ApplicationType.EARLY_WITHDRAWAL,
  );

const mapStateToProps = (state: State) => ({
  loading: state.login.loadingUserConversion,
  conversion: state.login.userConversion.secondPillar,
  sourceFunds: state.exchange.sourceFunds || [],
  targetFunds: state.exchange.targetFunds || [],
  secondPillarPikNumber: (state.login.user || {}).secondPillarPikNumber,
  secondPillarActive: (state.login.user || {}).secondPillarActive,
  secondPillarPaymentRate: (state.login.user || {}).secondPillarPaymentRate || 2,
});

export default connect(mapStateToProps)(SecondPillarStatusBox);
