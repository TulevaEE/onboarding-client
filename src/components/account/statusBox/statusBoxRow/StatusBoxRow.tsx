import React from 'react';
import './StatusBoxRow.scss';

const CheckMark: React.FunctionComponent<{
  checked?: boolean;
  warning?: boolean;
  error?: boolean;
}> = ({ checked = false, warning = false, error = false }) => (
  <div
    className={`ml-3 mr-2 fa${checked ? ' fa-check' : ''}${error ? ' fa-times' : ''}${
      warning ? ' fa-exclamation-triangle' : ''
    }`}
  />
);

export const StatusBoxRow: React.FunctionComponent<{
  name?: React.ReactNode;
  lines?: React.ReactNode[];
  showAction?: boolean;
  ok?: boolean;
  warning?: boolean;
  error?: boolean;
  last?: boolean;
  children?: React.ReactNode;
}> = ({
  name = '',
  lines = [],
  showAction = false,
  ok = false,
  warning = false,
  error = false,
  children = '',
  last = false,
}) => {
  const formattedLines = (
    <ul>
      {lines &&
        lines.length > 0 &&
        lines.map((line, i) => {
          return (
            <li className="pl-2" key={i}>
              {line}
            </li>
          );
        })}
    </ul>
  );
  return (
    <div
      className={`d-flex flex-sm-row flex-column justify-content-between py-2 status-box-row ${
        !last ? 'tv-table__row' : ''
      }`}
    >
      <div className="d-flex">
        <div className="d-flex flex-column justify-content-center">
          <CheckMark checked={ok} warning={warning} error={error} />
        </div>
        <div className="d-flex flex-column justify-content-center">
          <div className="mt-0 pt-1 pl-2">
            <b>{name}</b>
          </div>
          {formattedLines}
        </div>
      </div>
      <div className="d-flex flex-column justify-content-center my-2 mx-3 text-nowrap">
        {showAction && children ? children : ''}
      </div>
    </div>
  );
};

export default StatusBoxRow;
