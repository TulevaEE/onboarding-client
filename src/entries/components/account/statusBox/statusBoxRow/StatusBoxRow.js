import React from 'react';
import { PropTypes as Types } from 'prop-types';
import { Message } from 'retranslate';
import './StatusBoxRow.scss';

const CheckMark = ({ checked }) => (
  <div className={`ml-3 mr-2 fa${checked ? ' fa-check' : ' fa-times'}`} />
);
CheckMark.defaultProps = {
  checked: false,
};

CheckMark.propTypes = {
  checked: Types.bool,
};

const StatusBoxRow = ({ name, lines, showAction, ok, children, last }) => {
  const displayName = <Message>{name}</Message>;
  const formattedLines = (
    <ul className="status-box-row">
      {lines &&
        lines.length > 0 &&
        lines.map(line => {
          return (
            <li className="status-box-row pl-2" key={line}>
              {line}
            </li>
          );
        })}
    </ul>
  );
  return (
    <div className={`d-flex justify-content-between py-2 ${!last ? 'tv-table__row' : ''}`}>
      <div className="d-flex">
        <div className="d-flex flex-column justify-content-center">
          <CheckMark checked={ok} />
        </div>
        <div className="d-flex flex-column justify-content-center">
          <div className="mt-0 pt-1 pl-2">
            <b>{displayName}</b>
          </div>
          {formattedLines}
        </div>
      </div>
      <div className="d-flex flex-column justify-content-center mx-3 text-nowrap">
        {showAction && children ? children : ''}
      </div>
    </div>
  );
};

StatusBoxRow.defaultProps = {
  name: '',
  lines: [],
  showAction: false,
  ok: false,
  children: null,
  last: false,
};

StatusBoxRow.propTypes = {
  name: Types.oneOfType([Types.node, Types.string]),
  lines: Types.arrayOf(Types.oneOfType([Types.node, Types.string, Types.string])),
  showAction: Types.bool,
  ok: Types.bool,
  children: Types.node,
  last: Types.bool,
};

export default StatusBoxRow;
