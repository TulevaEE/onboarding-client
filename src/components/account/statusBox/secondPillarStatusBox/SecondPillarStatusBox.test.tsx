import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { Props, SecondPillarStatusBox } from './SecondPillarStatusBox';
import StatusBoxRow from '../statusBoxRow';
import {
  activeSecondPillar,
  completeSecondPillarConversion,
  highFeeSecondPillar,
  tulevaSecondPillarFund,
} from '../fixtures';

// TODO: Figure out a cleaner way to mock the hooks
jest.mock('../../../common/apiHooks', () => ({
  usePendingApplications: () => ({ data: [{ type: 'WITHDRAWAL' }] }),
  useMandateDeadlines: () => ({ data: { periodEnding: '2024-07-31T00:59:59.999999999Z' } }),
}));

jest.useFakeTimers();
jest.setSystemTime(new Date('2024-07-22T10:36:00Z'));

describe('SecondPillarStatusBox', () => {
  let component: ShallowWrapper;
  const props: Props = {
    loading: false,
    conversion: completeSecondPillarConversion.secondPillar,
    sourceFunds: [activeSecondPillar],
    targetFunds: [tulevaSecondPillarFund],
    secondPillarPikNumber: null,
    secondPillarActive: true,
    secondPillarPaymentRate: 6,
  };

  beforeEach(() => {
    component = shallow(<SecondPillarStatusBox {...props} />);
  });

  it('renders the success flow', () => {
    expect(component).toMatchSnapshot();
  });

  it('renders the payment rate flow', () => {
    component.setProps({ secondPillarPaymentRate: 2 });
    expect(component).toMatchSnapshot();
  });

  it('renders no 2nd pillar flow when no second pillar', () => {
    component.setProps({ secondPillarActive: false });
    expect(component).toMatchSnapshot();
  });

  it('does not show action when still loading', () => {
    component.setProps({ loading: true });
    expect(component.find(StatusBoxRow).prop('showAction')).toBeFalsy();
  });

  it('renders the withdrawal flow when withdrawal is in progress', () => {
    component.setProps({ conversion: { pendingWithdrawal: true } });
    expect(component).toMatchSnapshot();
  });

  it('renders low fee when fund selection incomplete', () => {
    component.setProps({
      conversion: { selectionPartial: false, selectionComplete: false, weightedAverageFee: 0.0049 },
    });
    expect(component).toMatchSnapshot();
  });

  it('renders high fee flow when fund selection incomplete', () => {
    component.setProps({
      sourceFunds: [highFeeSecondPillar],
      targetFunds: [tulevaSecondPillarFund],
      conversion: { selectionPartial: false, selectionComplete: false, weightedAverageFee: 0.0051 },
    });
    expect(component).toMatchSnapshot();
  });

  it('renders high fee message when in high fee fund and no partial conversion', () => {
    component.setProps({
      sourceFunds: [highFeeSecondPillar],
      targetFunds: [tulevaSecondPillarFund],
      conversion: {
        transfersPartial: false,
        transfersComplete: false,
        selectionPartial: false,
        selectionComplete: false,
        weightedAverageFee: 0.01,
      },
    });
    expect(component).toMatchSnapshot();
  });

  it('renders the transfer flow when fund transfers incomplete', () => {
    component.setProps({
      conversion: {
        transfersPartial: false,
        transfersComplete: false,
        selectionPartial: true,
        selectionComplete: true,
        weightedAverageFee: 0.01,
      },
    });
    expect(component).toMatchSnapshot();
  });

  it('renders the PIK flow when fund PIK number present', () => {
    component.setProps({ secondPillarPikNumber: 'EE1234567', sourceFunds: [] });
    expect(component).toMatchSnapshot();
  });
});
