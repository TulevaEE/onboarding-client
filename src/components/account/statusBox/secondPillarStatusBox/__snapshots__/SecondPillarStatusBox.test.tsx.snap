// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecondPillarStatusBox renders high fee flow when fund selection incomplete 1`] = `
<StatusBoxRow
  error={true}
  extraBottom={
    <div
      className="card card-primary border-0 mt-3 ml-3 ml-md-5 mr-3 my-2 px-4 py-3"
    >
      <div
        className="row"
      >
        <TinyCard
          img="euro.svg"
          text={
            <Memo(MemoizedFormattedMessage)
              id="account.status.choice.pillar.second.feecard.text"
              values={
                Object {
                  "b": [Function],
                }
              }
            />
          }
          title={
            <Memo(MemoizedFormattedMessage)
              id="account.status.choice.pillar.second.feecard.title"
            />
          }
        />
        <TinyCard
          img="basket.svg"
          text={
            <React.Fragment>
              <Memo(MemoizedFormattedMessage)
                id="account.status.choice.pillar.second.basketcard.text"
                values={
                  Object {
                    "b": [Function],
                  }
                }
              />
              <InfoTooltip
                name="diversification-tooltip"
              >
                <Memo(MemoizedFormattedMessage)
                  id="account.status.choice.pillar.second.basketcard.tooltip"
                />
              </InfoTooltip>
            </React.Fragment>
          }
          title={
            <Memo(MemoizedFormattedMessage)
              id="account.status.choice.pillar.second.basketcard.title"
            />
          }
        />
        <TinyCard
          img="deadline.svg"
          text={
            <Memo(MemoizedFormattedMessage)
              id="account.status.choice.pillar.second.deadlinecard.text"
              values={
                Object {
                  "b": [Function],
                  "periodEnding": "July 31 at 00:59",
                }
              }
            />
          }
          title={
            <Memo(MemoizedFormattedMessage)
              id="account.status.choice.pillar.second.deadlinecard.title"
            />
          }
        />
      </div>
    </div>
  }
  lines={
    Array [
      <React.Fragment>
        <Memo(MemoizedFormattedMessage)
          id="account.status.choice.highFee.label"
        />
        <InfoTooltip
          name="second-pillar-tooltip"
        >
          <Memo(MemoizedFormattedMessage)
            id="account.status.choice.highFee.description"
          />
        </InfoTooltip>
      </React.Fragment>,
      <small
        className="text-muted"
      >
        <Memo(MemoizedFormattedMessage)
          id="account.status.choice.highFee.comment"
          values={
            Object {
              "currentFeesEuro": <Euro
                amount={102.00000000000001}
                className="text-danger"
                fractionDigits={0}
              />,
              "tulevaFeesEuro": <Euro
                amount={98}
                className="text-success"
                fractionDigits={0}
              />,
            }
          }
        />
      </small>,
    ]
  }
  name={
    <Memo(MemoizedFormattedMessage)
      id="account.status.choice.pillar.second"
    />
  }
  showAction={true}
>
  <SecondPillarActionButton
    className="btn-primary"
  />
</StatusBoxRow>
`;

exports[`SecondPillarStatusBox renders high fee message when in high fee fund and no partial conversion 1`] = `
<StatusBoxRow
  error={true}
  extraBottom={
    <div
      className="card card-primary border-0 mt-3 ml-3 ml-md-5 mr-3 my-2 px-4 py-3"
    >
      <div
        className="row"
      >
        <TinyCard
          img="euro.svg"
          text={
            <Memo(MemoizedFormattedMessage)
              id="account.status.choice.pillar.second.feecard.text"
              values={
                Object {
                  "b": [Function],
                }
              }
            />
          }
          title={
            <Memo(MemoizedFormattedMessage)
              id="account.status.choice.pillar.second.feecard.title"
            />
          }
        />
        <TinyCard
          img="basket.svg"
          text={
            <React.Fragment>
              <Memo(MemoizedFormattedMessage)
                id="account.status.choice.pillar.second.basketcard.text"
                values={
                  Object {
                    "b": [Function],
                  }
                }
              />
              <InfoTooltip
                name="diversification-tooltip"
              >
                <Memo(MemoizedFormattedMessage)
                  id="account.status.choice.pillar.second.basketcard.tooltip"
                />
              </InfoTooltip>
            </React.Fragment>
          }
          title={
            <Memo(MemoizedFormattedMessage)
              id="account.status.choice.pillar.second.basketcard.title"
            />
          }
        />
        <TinyCard
          img="deadline.svg"
          text={
            <Memo(MemoizedFormattedMessage)
              id="account.status.choice.pillar.second.deadlinecard.text"
              values={
                Object {
                  "b": [Function],
                  "periodEnding": "July 31 at 00:59",
                }
              }
            />
          }
          title={
            <Memo(MemoizedFormattedMessage)
              id="account.status.choice.pillar.second.deadlinecard.title"
            />
          }
        />
      </div>
    </div>
  }
  lines={
    Array [
      <React.Fragment>
        <Memo(MemoizedFormattedMessage)
          id="account.status.choice.highFee.label"
        />
        <InfoTooltip
          name="second-pillar-tooltip"
        >
          <Memo(MemoizedFormattedMessage)
            id="account.status.choice.highFee.description"
          />
        </InfoTooltip>
      </React.Fragment>,
      <small
        className="text-muted"
      >
        <Memo(MemoizedFormattedMessage)
          id="account.status.choice.highFee.comment"
          values={
            Object {
              "currentFeesEuro": <Euro
                amount={200}
                className="text-danger"
                fractionDigits={0}
              />,
              "tulevaFeesEuro": <Euro
                amount={98}
                className="text-success"
                fractionDigits={0}
              />,
            }
          }
        />
      </small>,
    ]
  }
  name={
    <Memo(MemoizedFormattedMessage)
      id="account.status.choice.pillar.second"
    />
  }
  showAction={true}
>
  <SecondPillarActionButton
    className="btn-primary"
  />
</StatusBoxRow>
`;

exports[`SecondPillarStatusBox renders low fee when fund selection incomplete 1`] = `
<StatusBoxRow
  lines={
    Array [
      <Memo(MemoizedFormattedMessage)
        id="account.status.choice.lowFee.2.label"
        values={
          Object {
            "paymentRate": 2,
          }
        }
      />,
      <small
        className="text-muted"
      >
        <Memo(MemoizedFormattedMessage)
          id="account.status.choice.lowFee.index.2.description"
          values={
            Object {
              "paymentRate": 6,
              "paymentRateFulfillmentDate": "...",
            }
          }
        />
      </small>,
    ]
  }
  name={
    <Memo(MemoizedFormattedMessage)
      id="account.status.choice.pillar.second"
    />
  }
  ok={true}
  showAction={true}
>
  <SecondPillarActionButton
    className="btn-outline-primary"
  />
</StatusBoxRow>
`;

exports[`SecondPillarStatusBox renders no 2nd pillar flow when no second pillar 1`] = `
<StatusBoxRow
  error={true}
  lines={
    Array [
      <Memo(MemoizedFormattedMessage)
        id="account.status.choice.pillar.second.missing.label"
      />,
    ]
  }
  name={
    <Memo(MemoizedFormattedMessage)
      id="account.status.choice.pillar.second"
    />
  }
  showAction={true}
/>
`;

exports[`SecondPillarStatusBox renders the payment rate flow 1`] = `
<StatusBoxRow
  lines={
    Array [
      <Memo(MemoizedFormattedMessage)
        id="account.status.choice.lowFee.index.2.label"
        values={
          Object {
            "paymentRate": 2,
          }
        }
      />,
      <small
        className="text-muted"
      >
        <Memo(MemoizedFormattedMessage)
          id="account.status.choice.pillar.second.paymentRate.comment"
          values={
            Object {
              "b": [Function],
            }
          }
        />
      </small>,
    ]
  }
  name={
    <Memo(MemoizedFormattedMessage)
      id="account.status.choice.pillar.second"
    />
  }
  ok={true}
  showAction={true}
>
  <Link
    className="btn btn-primary"
    to="/2nd-pillar-payment-rate"
  >
    <MemoizedFormattedMessage
      id="account.status.choice.paymentRate.increase"
    />
  </Link>
</StatusBoxRow>
`;

exports[`SecondPillarStatusBox renders the success flow 1`] = `
<StatusBoxRow
  lines={
    Array [
      <Memo(MemoizedFormattedMessage)
        id="account.status.choice.lowFee.index.2.label"
        values={
          Object {
            "paymentRate": 2,
          }
        }
      />,
      <small
        className="text-muted"
      >
        <Memo(MemoizedFormattedMessage)
          id="account.status.choice.lowFee.index.2.description"
          values={
            Object {
              "paymentRate": 6,
              "paymentRateFulfillmentDate": "...",
            }
          }
        />
      </small>,
    ]
  }
  name={
    <Memo(MemoizedFormattedMessage)
      id="account.status.choice.pillar.second"
    />
  }
  ok={true}
  showAction={true}
/>
`;

exports[`SecondPillarStatusBox renders the transfer flow when fund transfers incomplete 1`] = `
<StatusBoxRow
  error={true}
  extraBottom={
    <div
      className="card card-primary border-0 mt-3 ml-3 ml-md-5 mr-3 my-2 px-4 py-3"
    >
      <div
        className="row"
      >
        <TinyCard
          img="euro.svg"
          text={
            <Memo(MemoizedFormattedMessage)
              id="account.status.choice.pillar.second.feecard.text"
              values={
                Object {
                  "b": [Function],
                }
              }
            />
          }
          title={
            <Memo(MemoizedFormattedMessage)
              id="account.status.choice.pillar.second.feecard.title"
            />
          }
        />
        <TinyCard
          img="basket.svg"
          text={
            <React.Fragment>
              <Memo(MemoizedFormattedMessage)
                id="account.status.choice.pillar.second.basketcard.text"
                values={
                  Object {
                    "b": [Function],
                  }
                }
              />
              <InfoTooltip
                name="diversification-tooltip"
              >
                <Memo(MemoizedFormattedMessage)
                  id="account.status.choice.pillar.second.basketcard.tooltip"
                />
              </InfoTooltip>
            </React.Fragment>
          }
          title={
            <Memo(MemoizedFormattedMessage)
              id="account.status.choice.pillar.second.basketcard.title"
            />
          }
        />
        <TinyCard
          img="deadline.svg"
          text={
            <Memo(MemoizedFormattedMessage)
              id="account.status.choice.pillar.second.deadlinecard.text"
              values={
                Object {
                  "b": [Function],
                  "periodEnding": "July 31 at 00:59",
                }
              }
            />
          }
          title={
            <Memo(MemoizedFormattedMessage)
              id="account.status.choice.pillar.second.deadlinecard.title"
            />
          }
        />
      </div>
    </div>
  }
  lines={
    Array [
      <React.Fragment>
        <Memo(MemoizedFormattedMessage)
          id="account.status.choice.highFee.label"
        />
        <InfoTooltip
          name="second-pillar-tooltip"
        >
          <Memo(MemoizedFormattedMessage)
            id="account.status.choice.highFee.description"
          />
        </InfoTooltip>
      </React.Fragment>,
      <small
        className="text-muted"
      >
        <Memo(MemoizedFormattedMessage)
          id="account.status.choice.highFee.comment"
          values={
            Object {
              "currentFeesEuro": <Euro
                amount={200}
                className="text-danger"
                fractionDigits={0}
              />,
              "tulevaFeesEuro": <Euro
                amount={98}
                className="text-success"
                fractionDigits={0}
              />,
            }
          }
        />
      </small>,
    ]
  }
  name={
    <Memo(MemoizedFormattedMessage)
      id="account.status.choice.pillar.second"
    />
  }
  showAction={true}
>
  <SecondPillarActionButton
    className="btn-primary"
  />
</StatusBoxRow>
`;

exports[`SecondPillarStatusBox renders the withdrawal flow when withdrawal is in progress 1`] = `
<StatusBoxRow
  error={true}
  lines={
    Array [
      <Memo(MemoizedFormattedMessage)
        id="account.status.choice.pillar.second.withdraw"
      />,
    ]
  }
  name={
    <Memo(MemoizedFormattedMessage)
      id="account.status.choice.pillar.second"
    />
  }
  showAction={true}
>
  <Link
    className="btn btn-primary"
    to="/applications/undefined/cancellation"
  >
    <MemoizedFormattedMessage
      id="account.status.choice.pillar.second.withdraw.cancel"
    />
  </Link>
</StatusBoxRow>
`;
